charts/node-analyzer/templates/_helpers.tpl

```diff
@@ -0,0 +1,166 @@
+{{- end -}}
+
+{{/*
+Return the proper image name for the Image Analyzer
+*/}}
+{{- define "nodeAnalyzer.image.imageAnalyzer" -}}
+    {{- include "nodeAnalyzer.imageRegistry" . -}} / {{- .Values.nodeAnalyzer.imageAnalyzer.image.repository -}} {{- if .Values.nodeAnalyzer.imageAnalyzer.image.digest -}} @ {{- .Values.nodeAnalyzer.imageAnalyzer.image.digest -}} {{- else -}} : {{- .Values.nodeAnalyzer.imageAnalyzer.image.tag -}} {{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper image name for the Host Analyzer
+*/}}
+{{- define "nodeAnalyzer.image.hostAnalyzer" -}}
+    {{- include "nodeAnalyzer.imageRegistry" . -}} / {{- .Values.nodeAnalyzer.hostAnalyzer.image.repository -}} {{- if .Values.nodeAnalyzer.hostAnalyzer.image.digest -}} @ {{- .Values.nodeAnalyzer.hostAnalyzer.image.digest -}} {{- else -}} : {{- .Values.nodeAnalyzer.hostAnalyzer.image.tag -}} {{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper image name for the Benchmark Runner
+*/}}
+{{- define "nodeAnalyzer.image.benchmarkRunner" -}}
+    {{- include "nodeAnalyzer.imageRegistry" . -}} / {{- .Values.nodeAnalyzer.benchmarkRunner.image.repository -}} {{- if .Values.nodeAnalyzer.benchmarkRunner.image.digest -}} @ {{- .Values.nodeAnalyzer.benchmarkRunner.image.digest -}} {{- else -}} : {{- .Values.nodeAnalyzer.benchmarkRunner.image.tag -}} {{- end -}}
+{{- end -}}
+
+{{/*
+Node Analyzer labels
+*/}}
+{{- define "nodeAnalyzer.labels" -}}
+helm.sh/chart: {{ include "nodeAnalyzer.chart" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+  {{- if .Values.nodeAnalyzer.labels }}
+    {{- $tp := typeOf .Values.nodeAnalyzer.labels }}
+    {{- if eq $tp "string" }}
+      {{- tpl .Values.nodeAnalyzer.labels . }}
+    {{- else }}
+      {{- toYaml .Values.nodeAnalyzer.labels }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{- define "deploy-nia" -}}
+{{- if or .Values.nodeAnalyzer.runtimeScanner.deploy .Values.nodeAnalyzer.runtimeScanner.settings.collectorEndpoint -}}
+true
+{{- end -}}
+{{- end -}}
+
+{{/*
+Sysdig Eve Connector service URL
+*/}}
+{{- define "eveconnector.host" -}}
+{{ include "nodeAnalyzer.fullname" .}}-eveconnector.{{ .Release.Namespace }}
+{{- end -}}
+
+{{/*
+Sysdig Eve Connector Secret generation (if not exists)
+*/}}
+{{- define "eveconnector.token" -}}
+{{- $secret := lookup "v1" "Secret" .Release.Namespace "sysdig-eve-secret" -}}
+{{- if $secret -}}
+{{ $secret.data.token }}
+{{- else -}}
+{{ randAlphaNum 32 | b64enc | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+The following helper functions are all designed to use global values where
+possible, but accept overrides from the chart values.
+*/}}
+{{- define "nodeAnalyzer.accessKey" -}}
```

Comment:
New addition over the sysdig chart.
This helper function is to make use of global values wherever possible.